### ISSUE TITLE
The closing delimiter of a div block can interrupt a paragraph

### ISSUE THREAD

**Issue by rauschma** [jgm/djot#291]
Markdown:

```
::: note
line
:::
```

HTML:

```html
<div class="note">
<p>line</p>
</div>
```

[Playground](https://djot.net/playground/?text=%3A%3A%3A+note%0Aline%0A%3A%3A%3A&mode=html&sourcepos=false)

Doesn’t this conflict with the rule “Block-level elements can't interrupt paragraphs”?

**Comment by jgm** [comment]
Not really; this isn't a new block-level element, it's just the end of one that had already been started.

**Comment by rauschma** [comment]
OK. I’m torn:
* Div blocks look nice this way.
* The usability argument for paragraphs doesn’t change if it’s a closing block delimiter (vs. an opening block delimiter).

**Comment by jgm** [comment]
Philosophically, I would almost prefer not to have any fenced constructs for this reason.

A side-marked div construct (like blockquotes, but with `:`) has several advantages:  (a) it is *local* (you don't need to look potentially far back in the file to see if you're in a div), (b) nesting is completely clear, (c) it doesn't have this problem with the end fence, (d) it is more consistent, since the other block containers are non-fenced (block quotes, list items).

When I've brought this up, though, in commonmark and pandoc forums, people have consistently preferred fenced syntax.  I'm still not convinced, though.  This could be brought up separately on the discussion list.

**Comment by rauschma** [comment]
Interesting!

Visually, it helps that closing delimiters have to match the colon counts of opening delimiters:

```md
::::::

Outer

:::
Inner
:::
::::::
```

I’m currently migrating my parser from CommonMark-ish to Djot and it’s been a big improvement: I especially love how Djot uses the column number of the opening code block delimiter to determine where code block lines begin.

The only exception was that the “paragraphs can’t be interrupted” rule made two constructs more difficult to use:

* Lists: empty line required before nested lists
* Div blocks: can’t always write them as tightly as I’d like (see below).

```md
::::::
Outer
:::
Inner
:::
::::::
```
